File: v18_decorator/java/Coffee1.java
package v18_decorator.java;

interface Coffee {
    double cost();
}

class SimpleCoffee implements Coffee {
    public double cost() {
        return 2;
    }
}

class CoffeeWithMilk implements Coffee {
    public double cost() {
        return 2 + 0.5;
    }
}

class CoffeeWithSugar implements Coffee {
    public double cost() {
        return 2 + 0.25;
    }
}

class CoffeeWithMilkAndSugar implements Coffee {
    public double cost() {
        return 2 + 0.5 + 0.25;
    }
}

File: v18_decorator/python/coffee1.py
from abc import ABC, abstractmethod

class Coffee(ABC):
    @abstractmethod
    def cost(self):
        pass

class SimpleCoffee(Coffee):
    def cost(self):
        return 2

class CoffeeWithMilk(Coffee):
    def cost(self):
        return 2 + 0.5

class CoffeeWithSugar(Coffee):
    def cost(self):
        return 2 + 0.25

class CoffeeWithMilkAndSugar(Coffee):
    def cost(self):
        return 2 + 0.5 + 0.25

